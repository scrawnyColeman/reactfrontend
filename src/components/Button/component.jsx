import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import { default as MaterialButton } from '@material-ui/core/Button';
import { colours, materialStyles } from '../../constants/styles';

const Button = ({ text, hierarchy = 'primary', size = 'large', onClick }) => {
    const useStyles = makeStyles(materialStyles);
    const classes = useStyles();
    const buttonOverrideStyles = {
        fontSize: size === 'large' ? '1.25rem' : '1rem',
        textTransform: 'unset',
        color: colours.primary,
    };
    return (
        <MaterialButton
            color={hierarchy}
            size={size}
            className={classes.button}
            style={buttonOverrideStyles}
            variant="text"
            onClick={onClick}
        >
            {text}
        </MaterialButton>
    );
};

export default Button;
