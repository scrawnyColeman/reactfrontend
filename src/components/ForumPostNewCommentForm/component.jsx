import React, { useState } from 'react';
import styled from 'styled-components';
import TextField from '../TextField/component';
import Button from '../Button/component';

const Container = styled.div`
    text-align: left;
`;
const StyledHeading = styled.div`
    padding: 0 0 14px;
    font-size: 2rem;
`;
const StyledWrapper = styled.div`
    display: grid;
    grid-template-rows: min-content max-content;
    grid-gap: 8px;
    margin: 0 auto;
`;

const ForumPostNewCommentForm = ({ isReplying, replyingTo }) => {
    const [commentUser, setCommentUser] = useState('default value');
    const [question, setQuestion] = useState(null);

    return (
        <Container>
            <StyledHeading>Join the discussion...</StyledHeading>
            <StyledWrapper>
                {!isReplying ? (
                    <TextField
                        id="outlined-full-width"
                        label="Label"
                        margin="normal"
                        onChange={(event) => setCommentUser(event.target.value)}
                        disabled
                        placeholder={'New Comment'}
                    />
                ) : (
                    <TextField
                        id="outlined-full-width"
                        label="Label"
                        margin="normal"
                        onChange={(event) => setCommentUser(event.target.value)}
                        disabled
                        placeholder={`Replying to: ${replyingTo}`}
                    />
                )}
                <TextField
                    id="filled-multiline-static"
                    text={question}
                    margin="normal"
                    rows={20}
                    multiline={true}
                    onChange={(event) => setQuestion(event.target.value)}
                    placeholder={'Write your comment here...'}
                />
                <Button
                    onClick={() => window.location.pathname === '/'}
                    text="Submit"
                    size="large"
                    variant="outlined"
                    hierarchy="primary"
                />
            </StyledWrapper>
        </Container>
    );
};

export default ForumPostNewCommentForm;
