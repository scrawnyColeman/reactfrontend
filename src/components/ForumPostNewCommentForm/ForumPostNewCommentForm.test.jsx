import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import 'jest-styled-components';
import React from 'react';
import ForumPostNewCommentForm from './component';
import Button from '../Button/component';

Enzyme.configure({ adapter: new Adapter() });

global.window = { location: { pathname: '/hello' } };

describe('ForumPostNewCommentForm component', () => {
    test('component matches snapshot', () => {
        const wrapper = shallow(<ForumPostNewCommentForm />);
        expect(wrapper).toMatchSnapshot();
    });

    test('Button click updates window pathname', () => {
        const wrapper = shallow(<ForumPostNewCommentForm />);
        wrapper.find(Button).simulate('click');

        expect(global.window.location.pathname).toBe('/');
    });
});
