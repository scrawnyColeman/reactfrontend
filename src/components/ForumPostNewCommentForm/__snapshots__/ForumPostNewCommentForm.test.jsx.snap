// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForumPostNewCommentForm component component matches snapshot 1`] = `
<styled.div>
  <styled.div>
    Join the discussion...
  </styled.div>
  <styled.div>
    <TextField
      disabled={true}
      id="outlined-full-width"
      label="Label"
      margin="normal"
      onChange={[Function]}
      placeholder="New Comment"
    />
    <TextField
      id="filled-multiline-static"
      margin="normal"
      multiline={true}
      onChange={[Function]}
      placeholder="Write your comment here..."
      rows={20}
      text={null}
    />
    <Button
      hierarchy="primary"
      onClick={[Function]}
      size="large"
      text="Submit"
      variant="outlined"
    />
  </styled.div>
</styled.div>
`;
